docs/source/dlv2_tutorial.rst

```diff
@@ -0,0 +1,52 @@
+DataLoader2 Tutorial
+=====================
+
+This is the tutorial for users to create ``DataPipe`` graph and load data via ``DataLoader2`` with different backend systems (``ReadingService``).
+
+DataPipe
+---------
+
+Please refer to `DataPipe Tutorial <dp_tutorial.html>_` for more details. Here are the most important caveats:
+to make sure the data pipeline has different order per epoch and data shards are mutually exclusive and collectively exhaustive:
+
+- Place ``sharding_filter`` or ``sharding_round_robin_dispatch`` as early as possibel in the pipele to avoid repeating expensive operations in worker/distributed processes.
+- Add a ``shuffle`` DataPipe before sharding to achieve inter-shard shuffling. ``ReadingService`` will handle synchronization of those ``shuffle`` operations to the order of data are the same before sharding so that all shards are mutually exclusive and collectively exhaustive.
```

Comment:
```suggestion
- Add a ``shuffle`` DataPipe before sharding to achieve inter-shard shuffling. ``ReadingService`` will handle synchronization of those ``shuffle`` operations to ensure the order of data are the same before sharding so that all shards are mutually exclusive and collectively exhaustive.
```
This change should suffice but consider rewording the sentence